pkg/prometheus/agent/operator.go

```diff
@@ -177,62 +119,32 @@ func New(ctx context.Context, conf operator.Config, logger log.Logger, r prometh
 		return nil, errors.Wrap(err, "error creating prometheus-agent informers")
 	}
 
+	c.metrics.MustRegister(
+		c.reconciliations,
+	)
+
+	c.rr = operator.NewResourceReconciler(
+		c.logger,
+		c,
+		c.metrics,
+		monitoringv1alpha1.PrometheusAgentsKind,
+		r,
+	)
+
 	var promStores []cache.Store
 	for _, informer := range c.promInfs.GetInformers() {
 		promStores = append(promStores, informer.Informer().GetStore())
 	}
 
 	c.metrics.MustRegister(prompkg.NewCollectorForStores(promStores...))
 
-	c.smonInfs, err = informers.NewInformersForResource(
-		informers.NewMonitoringInformerFactories(
-			c.config.Namespaces.AllowList,
-			c.config.Namespaces.DenyList,
-			mclient,
-			resyncPeriod,
-			nil,
-		),
-		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.ServiceMonitorName),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating servicemonitor informers")
-	}
-
-	c.pmonInfs, err = informers.NewInformersForResource(
-		informers.NewMonitoringInformerFactories(
-			c.config.Namespaces.AllowList,
-			c.config.Namespaces.DenyList,
-			mclient,
-			resyncPeriod,
-			nil,
-		),
-		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.PodMonitorName),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating podmonitor informers")
-	}
-
-	c.probeInfs, err = informers.NewInformersForResource(
-		informers.NewMonitoringInformerFactories(
-			c.config.Namespaces.AllowList,
-			c.config.Namespaces.DenyList,
-			mclient,
-			resyncPeriod,
-			nil,
-		),
-		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.ProbeName),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating probe informers")
-	}
-
 	if c.scrapeConfigSupported {
 		c.sconInfs, err = informers.NewInformersForResource(
```

Comment:
Do you want to centralize the creation of the ScrapeConfig informer as well or keep it as a follow up? (see #5545)

Reply:
followup make sense, as this PR become a huge, especially with rebasing 😄